specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/preview/2018-08-15-preview/timeseriesinsights.json

```diff
@@ -211,7 +211,7 @@
         ],
         "responses": {
           "200": {
-            "description": "The environment definition was successfully updated and is in the response. If the environment was updated synchronously, the response will include a provisioningState value of \"Succeeded\". If the environment was updated asynchronous, the response will include a provisioningState value of \"Updating\".  You can periodically get your environment definition and monitor progress of the update via the provisioningState property.",
+            "description": "The environment definition was successfully updated and is in the response. If the environment was updated synchronously, the response will include a provisioningState value of \"Succeeded\". If the environment was updated asynchronously, the response will include a provisioningState value of \"Updating\".  You can periodically get your environment definition and monitor progress of the update via the provisioningState property.",
```

Comment:
Has this been fixed on service side accordingly? 

This one succeeds with 200, without any indication this is a Long Running Operation. Note that `"provisioningState": "Succeeded",`

```
az timeseriesinsights environment longterm update --resource-group rg1 --name "env2" --data-retention 12 --debug

msrest.http_logger : Request URL: 'https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/rg1/providers/Microsoft.TimeSeriesInsights/environments/env2?api-version=2018-08-15-preview'
msrest.http_logger : Request method: 'PATCH'
msrest.http_logger : Request headers:
msrest.http_logger :     'Accept': 'application/json'
msrest.http_logger :     'Content-Type': 'application/json; charset=utf-8'
msrest.http_logger :     'accept-language': 'en-US'
msrest.http_logger :     'Content-Length': '69'
msrest.http_logger :     'User-Agent': 'python/3.8.1 (Windows-10-10.0.18362-SP0) msrest/0.6.10 msrest_azure/0.6.3 azure-mgmt-timeseriesinsights/0.2.0 Azure-SDK-For-Python AZURECLI/2.4.0 (PIP)'
msrest.http_logger : Request body:
msrest.http_logger : {"properties": {"warmStoreConfiguration": {"dataRetention": "P12D"}}}

msrest.http_logger : Response status: 200
msrest.http_logger : Response headers:
msrest.http_logger :     'Cache-Control': 'no-cache'
msrest.http_logger :     'Pragma': 'no-cache'
msrest.http_logger :     'Transfer-Encoding': 'chunked'
msrest.http_logger :     'Content-Type': 'application/json'
msrest.http_logger :     'Content-Encoding': 'gzip'
msrest.http_logger :     'Expires': '-1'
msrest.http_logger :     'Vary': 'Accept-Encoding'
msrest.http_logger :     'x-ms-ratelimit-remaining-subscription-writes': '1196'
msrest.http_logger :     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
msrest.http_logger :     'X-Content-Type-Options': 'nosniff'
msrest.http_logger :     'x-ms-request-id': '6dfebd67-dd7f-4798-87ee-9c35fa2f934f'
msrest.http_logger :     'Server': 'Microsoft-HTTPAPI/2.0'
msrest.http_logger :     'x-ms-correlation-request-id': 'e0d68eec-428d-40de-8849-24746a395303'
msrest.http_logger :     'x-ms-routing-request-id': 'SOUTHEASTASIA:20200423T080138Z:e0d68eec-428d-40de-8849-24746a395303'
msrest.http_logger :     'Date': 'Thu, 23 Apr 2020 08:01:37 GMT'
msrest.http_logger : Response content:
msrest.http_logger : 
{
    "sku": {
        "name": "L1",
        "capacity": 1
    },
    "kind": "LongTerm",
    "location": "WestUs",
    "tags": {},
    "properties": {
        "timeSeriesIdProperties": [
            {
                "name": "['DeviceId1']",
                "type": "String"
            }
        ],
        "storageConfiguration": {
            "accountName": "st0423"
        },
        "warmStoreConfiguration": {
            "dataRetention": "P12D"
        },
        "creationTime": "Thu, 23 Apr 2020 07:31:28 GMT",
        "provisioningState": "Succeeded",
        "requestApiVersion": "2018-08-15-preview",
        "dataAccessId": "7edc94e2-0d50-45ec-87f0-8581f9fdb896",
        "dataAccessFqdn": "7edc94e2-0d50-45ec-87f0-8581f9fdb896.env.timeseries.azure.com"
    },
    "id": "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/rg1/providers/Microsoft.TimeSeriesInsights/environments/env2",
    "name": "env2",
    "type": "Microsoft.TimeSeriesInsights/Environments"
}
```

Then the following command fails because the previous operation hasn’t finished.

```
az timeseriesinsights environment longterm update --resource-group rg1 --name "env2" --data-retention 14 --debug

msrest.http_logger : Request URL: 'https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/rg1/providers/Microsoft.TimeSeriesInsights/environments/env2?api-version=2018-08-15-preview'
msrest.http_logger : Request method: 'PATCH'
msrest.http_logger : Response status: 409
msrest.http_logger : Response content:
msrest.http_logger : {"error":{"code":"EnvironmentOperationInProgress","message":"An operation is already in progress on the environment."}}
```

Reply:
Hi Jiashuo,

We have no record of this issue on our tracks. Would you mind opening an issue for us to investigate, fix and keep track of it, please? You can assign it directly to me.

We're going to add this to our backlog.

Reply:
Thank you @riserrad, I have created issue https://github.com/Azure/azure-rest-api-specs/issues/9265.